addons/account/models/sequence_mixin.py

```diff
@@ -45,20 +45,50 @@ def _check_grouping(grouping, optional=None, required=None):
             return 'never'
         raise ValidationError(_('The sequence regex should at least contain the prefix1, seq and suffix grouping keys. For instance:\n^(?P<prefix1>.*?)(?P<seq>\d*)(?P<suffix>\D*?)$'))
 
+    def _constrains_regex_match(self):
+        try:
+            name = None
+            for rec in self:
+                name = rec[rec._sequence_field]
+                rec._deduce_sequence_number_reset(name)
+        except ValidationError:
+            raise ValidationError(_("Field %s should have a numeric part. Got %r") % (rec._sequence_field, name)) from None
+
+    def __init__(self, pool, cr):
+        api.constrains(self._sequence_field)(type(self)._constrains_regex_match)
+        return super().__init__(pool, cr)
+
+    def _reset_mode_to_order(self, mode):
```

Comment:
I'd like `_regex_to_order` or something like that better, you are reworking the regex, not the reset mode.

Reply:
I started with this idea but using the reset mode as entry leads to less API change.
This would need that `_deduce_sequence_number_reset` should also return the matched regex, which is weird.